asm/kanread.s

```diff
@@ -0,0 +1,209 @@
+.include "macro.inc"
+#include "ultra64/asm.h"
+
+// assembler directives
+.set noat
+.set noreorder
+.set gp=64
+
+.section .text
+
+.balign 16
+
+//! s32 Kanji_OffsetFromShiftJIS(s32 shiftJISCodepoint);
+//! 
+//! Returns the offset of the glyph texture data in the file `kanji` corresponding
+//! to a given 2-byte Shift-JIS codepoint. No range validity check is carried out.
+//!
+//! A nice Shift-JIS codepoint table: https://uic.io/en/charset/show/shift_jis/
+//! The file `kanji` contains the 'Level 1' kanji (0x889F-0x9872), and a reworked
+//! version of the non-kanji section that includes extra English and Hylian glyphs.
+//!
+//! @param shiftJISCodepoint Codepoint of glyph.
+//! return s32 offset into `kanji` file.
+//!
+//! @remark Original name: "LeoGetKadr"
+LEAF(Kanji_OffsetFromShiftJIS)
+    // Characters with codepoints >= 0x8800 are kanji. Arrangement is regular, 
+    // so convert index directly.
+    li      $at, 0x8800
+    slt     $at, $a0, $at
+    bnez    $at, .nonkanji
+     // 0xBC is number of glyphs in one block in the `kanji` file:
+     // 0x100 possible codepoints with the same byte1
+     // - 0x40 unused at beginning 
+     // - 1 unused at 0x7F 
+     // - 3 unused at 0xFD, 0xFE, 0xFF
+     li     $a2, 0xBC
```

Comment:
Would it help modding and/or readability if defines were made for this number? I'd also want to suggest a define for the texture size but since it's encoded in a shift I recognise that may not be as useful. Might still be nice to have the shift amount in one place rather than two though.